AndroidSDKCore/src/main/java/com/leanplum/internal/RequestOld.java

```diff
@@ -516,20 +520,29 @@ private void sendNow() {
 
     Leanplum.countAggregator().incrementCount("send_now");
 
-    Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
-      @Override
-      protected Void doInBackground(Void... params) {
-        try {
-          sendRequests();
-        } catch (Throwable t) {
-          Util.handleException(t);
+    // If we are running on main thread, fire the async task to execute the request async,
+    // otherwise send the request synchronously.
+    if (Looper.getMainLooper().getThread() == Thread.currentThread()) {
+      Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
+        @Override
+        protected Void doInBackground(Void... params) {
+          try {
+            sendRequests();
+          } catch (Throwable t) {
+            Util.handleException(t);
+          }
+          return null;
         }
-        return null;
+      });
+    } else {
+      try {
+        sendRequests();
```

Comment:
Both if and else do the same thing?           try {
            sendRequests();
          } catch (Throwable t) {
            Util.handleException(t);
          }

Reply:
Not really, the first one is offloading request into background task, the else part is running it on current thread (which is background thread)

Reply:
Can you please let me know why is that point above? Why couldnt we just run in background thread, no matter what the parent thread is?

Reply:
Yeah it doesn't really matter, we can always create new async task, but in that case I'll need to synchronise call to forceContentUpdate in LeanplumPushService, to make sure that IntentService doesn't get killed.

Reply:
I would prefer that approach please. Its way more cleaner than handling these If else conditions. Can you please make that change?